### PR TITLE
Update JS libraries with potential security vulnerabilities

### PR DESCRIPTION
**Pull request by biyeun**
https://dimagi-dev.atlassian.net/browse/SAAS-10882

##### SUMMARY
Updates knockout, jquery, bootstrap (3.x), and DOMPurify to their latest versions.

##### RISK ASSESSMENT / QA PLAN
I'm leaving this on staging for a day or two while I play around with things and see if there are any issues. Also alerting Ben.

### CODE REVIEW

File: corehq/apps/hqwebapp/templates/hqwebapp/partials/ko_inline_edit.html [hunk 1]
```diff
@@ -48,5 +48,5 @@
         </button>
       </div>
     </div>
-  </div><span data-bind="template: {afterRender: afterRenderFunc}"></span>
+  </div><span data-bind="template: {afterRender: afterRenderFunc}">&nbsp;</span> {# please keep the &nbsp; to satisfy the knockout 3.5.1 gods #}
```
Comment (orangejenny) [inline corehq/apps/hqwebapp/templates/hqwebapp/partials/ko_inline_edit.html:51]: Do you know which part requires the `&nbsp;`? I'm wondering if this problem could affect other usages of `template` or specifically `afterRender`.

File: corehq/apps/hqwebapp/static/hqwebapp/js/hq.helpers.js [hunk 2]
```diff
@@ -131,6 +131,7 @@ hqDefine("hqwebapp/js/hq.helpers", [
         if (this.length > 1) {
             throw new Error("Multiple elements passed to koApplyBindings");
         }
+        ko.cleanNode(this.get(0)); // make sure we don't apply bindings twice to the same element
```
Comment (orangejenny) [inline corehq/apps/hqwebapp/static/hqwebapp/js/hq.helpers.js:134]: Is there a change in knockout requiring this? The error about applying bindings twice often points to a real bug.

**Comment by orangejenny** [comment]
Were you able to read through release notes for these upgrades?

**Comment by biyeun** [comment]
@orangejenny yea i should have been clearer in the PR that this is a WIP, and I'll be pushing things to staging and pulling them to test things. I'm reading through the release notes—knockout has some pretty scary changes—and this is by no means ready to merge yet..

**Comment by orangejenny** [comment]
@biyeun got it - good luck!

**Pull request closed without merging**